app/vmselect/prometheus/prometheus.go

```diff
@@ -537,26 +538,44 @@ func resetRollupResultCaches() {
 			" For example: -selectNode=select-addr-1:8481,select-addr-2:8481")
 		return
 	}
+	var defaultPort uint16 = 8481
```

Comment:
I'd recommend to use the same discovery pattern that is used by vmauth for backend discovery.

Reply:
Sorry, I should be more specific:

It'd be great to add the following syntax support to the vmselectNode discovery (like vmauth does):
```
-selectNode=http://srv+node_srv_record:8431/expose_prefix
-selectNode=https://srv+node_srv_record:8431
-selectNode=direct-node-nodename
-selectNode=direct-node-nodename-1:8481
-selectNode=https://username:password@srv+node_srv_record_a:8431
```

 It should improve users experience with cache reset configuration and make it more flexible. 

 It also would be great to iterate over parsed select node urls:
`for _, snURL := range selectNodeURLs { `

something like: 
```go
for _, snURL := range selectNodeURLs {
               host := snURL.Hostname()
               port := snURL.Port()
		var resolvedAddrs []string
               if !strings.HasPrefix(host, "srv+") {
                        // fast path with old logic
                        resetRollupResultCacheForURL(snURL)
                        continue
                }
			// The host has the format 'srv+realhost'. Strip 'srv+' prefix before performing the lookup.
			srvHost := strings.TrimPrefix(host, "srv+")
			_, addrs, err := netutil.Resolver.LookupSRV(ctx, "", "", srvHost)
			if err != nil {
				logger.Warnf("cannot discover select node SRV records for %s: %s; use it literally", snURL, err)
				continue
			} 
                        resolvedAddrs = make([]string, len(addrs))
                        // add goroutine if needed
			for i, addr := range addrs {
                            snURLCopy := *snURL
                            snURLCopy.Host = addr.Target
                            if snURLCopy.Port == ""{
                                 // replace port only if it's not defined
                                  snURLCopy.Port = addr.Port
                            }
                            resetRollupResultCacheForURL(snURLCopy)
			}
}
```

Reply:
updated